File: Algorithms/Sorting Algorthms/Insertion Sort/InsertionSort.py
class InsertionSort:
    
    def printArray(self, arr):
        for i in arr:
            print(i, end=" ")
        print()
        
    def inSort(self, arr):
        n = len(arr)
        for i in range(1, n):
            currentMin = arr[i]
            j = i - 1
            while j >= 0 and arr[j] > currentMin:
                arr[j + 1] = arr[j]
                j -= 1
            arr[j + 1] = currentMin
        return arr 
  
if __name__ == "__main__":
    
    insertObj = InsertionSort()
    array = [2, 4, 5, 32, 56, 82, 11, 0, 75, 323, 5245, 74]
    
    print("Old array    -> ", end="")
    insertObj.printArray(array)
    
    SortedArray = insertObj.inSort(array)
    print("Sorted array -> ", end="")
    insertObj.printArray(SortedArray)

File: Algorithms/Sorting Algorthms/Insertion Sort/InsertionSort.java
public class InsertionSort {
    
    public static void main(String[] args) {
        int array [] = {2,4,1,6,5,7,9,4,90,235,642,46};

        System.out.print("old array    -> ");
        printArray(array);

        int [] Sorted = InSort(array);
        System.out.print("sorted array -> ");
        printArray(Sorted);
    }

    public static int[] InSort (int arr []){
        int n = arr.length;
        for( int i=  1; i<n ; i++ ){
            int j = i-1;
            int currenMin = arr[i];
            while (j>=0 && arr[j] > currenMin ){
                arr[j+1] = arr[j]; // swap
                j--;
            }
            arr[j+1] = currenMin;
        }
        return arr.clone();
    }

    public static void printArray (int arr[]){
        for (int i : arr){
            System.out.print(i + " ");
        }
        System.out.println();
    }
}
